### PR TITLE
Tma 424 fix s3 listing limit

### PR DESCRIPTION
**Pull request by Aidan23Smith**
Right now, if there are over 1000 objects in an S3 buckets, only the first 1000 are checked for the latest objects. However, the latest objects will not be in this batch. Also, if there are no events, the step definitions fail.

To account for this, we will search all the batches for the latest keys. Also, in the event-processor test, we will empty the bucket after each successful run through of the test to save on storage.

### CODE REVIEW

**Comment by jashelb90** [comment]
LGTM but will wait for Des due to his Java knowledge

File: tests/event-processing-tests/src/main/java/Cleanup.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import io.cucumber.java.en.And;
```
Comment (des-hippo) [inline tests/event-processing-tests/src/main/java/Cleanup.java:1]: Can we move this to an appropriate package directory, and add the package statement here. I know it’s only test code but best to keep it consistent.

**Review by des-hippo — DISMISSED** [review]
One small comment about packaging. Otherwise LGTM.

**Review by des-hippo — APPROVED** [review]
LGTM
